### PR TITLE
Proof missing errors returned in cases where empty proofs are expected

### PR DESCRIPTION
**Pull request by CHr15F0x**
Fixes: #2416 

Also fixes:
- no tests for `pathfinder_getClassProof`

### CODE REVIEW

**Review by vbar — APPROVED** [review]

**Comment by kkovaacs** [comment]
> "partial" queries for starknet_getStorageProof not possible for certain queries & conditions,

BTW, it would be probably worth adding an issue describing the exact cases where it did not work, and also add that to the CHANGELOG as this is a user-visible change.

Plus: can you maybe come up with a better name for the PR?

**Comment by CHr15F0x** [comment]
>BTW, it would be probably worth adding an issue describing the exact cases where it did not work, and also add that to the CHANGELOG as this is a user-visible change.

Definitely :+1: 

> Plus: can you maybe come up with a better name for the PR?

Sure :+1:

File: crates/rpc/src/method/get_storage_proof.rs [hunk 1]
```diff
@@ -988,6 +988,7 @@ mod tests {
         }
     }
 
+    /*
```
Comment (vbar) [inline crates/rpc/src/method/get_storage_proof.rs:991]: The test could be fixed, right?

Comment (CHr15F0x) [reply crates/rpc/src/method/get_storage_proof.rs]: Yes, it's a side effect of rework in progress, the PR was put back to draft at that point.

File: crates/rpc/src/method/get_storage_proof.rs [hunk 2]
```diff
@@ -456,10 +340,170 @@ pub async fn get_storage_proof(context: RpcContext, input: Input) -> Result<Outp
...
+    tx: &Transaction<'_>,
+    block_number: BlockNumber,
+    contract_addresses: &Option<Vec<ContractAddress>>,
+) -> Result<(Felt, NodeHashToNodeMappings, Vec<ContractLeafData>), Error> {
+    let Some(storage_root_idx) = tx
+        .storage_root_index(block_number)
+        .context("Querying storage root index")?
+    else {
+        if tx.trie_pruning_enabled() {
+            eprintln!("01");
```
Comment (kkovaacs) [inline crates/rpc/src/method/get_storage_proof.rs:399]: Debug print?

**Review by kkovaacs — COMMENTED** [review]
LGTM, except for the trivial comments I've added.

**Pull request merged**